app/src/test/java/com/example/plantsapp/data/usecase/GetTasksForPlantAndDateUseCaseImplTest.kt

```diff
@@ -0,0 +1,144 @@
+package com.example.plantsapp.data.usecase
+
+import com.example.plantsapp.data.repository.StubTasksHistoryRepository
+import com.example.plantsapp.data.repository.StubTasksRepository
+import com.example.plantsapp.data.utils.createPlant
+import com.example.plantsapp.domain.model.Task
+import com.example.plantsapp.domain.model.TaskWithState
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import java.util.Date
+import java.util.Calendar
+
+class GetTasksForPlantAndDateUseCaseImplTest {
+
+    private fun createDate(year: Int, month: Int, date: Int): Date {
+        return Calendar.getInstance().apply {
+            set(year, month, date)
+        }.time
+    }
+
+    private fun createAccurateDate(
+        year: Int,
+        month: Int,
+        date: Int,
+        hour: Int,
+        minute: Int
+    ): Date {
+        return Calendar.getInstance().apply {
+            set(year, month, date)
+            set(Calendar.HOUR_OF_DAY, hour)
+            set(Calendar.MINUTE, minute)
+        }.time
+    }
+
+    private fun createUseCase(tasksList: List<Task>): GetTasksForPlantAndDateUseCaseImpl {
+        return GetTasksForPlantAndDateUseCaseImpl(
+            tasksRepository = StubTasksRepository(tasksList),
+            tasksHistoryRepository = StubTasksHistoryRepository()
+        )
+    }
+
+    @Test
+    fun `tasks for creation date - returns all tasks`() = runBlockingTest {
+        val creationDate = createDate(2021, 11, 10)
+        val initialList = listOf(
+            Task.WateringTask(frequency = 11),
+            Task.SprayingTask(frequency = 2)
+        )
+        val useCase = createUseCase(initialList)
+
+        val actual = useCase
```

Comment:
maybe add normal name for use case here

Because "createUseCase" makeYou think that it is use case for creating something